crates/block-producer/src/domain/mod.rs

```diff
@@ -1 +1,2 @@
-pub mod transaction;
+pub(crate) mod batch;
+pub(crate) mod transaction;
```

Comment:
nit: I guess this technically makes this PR breaking. Probably zero impact, but just noting it

Reply:
We sort of hand-wavingly define our public API as the node's CLI, the public gRPC schema and any major protocol or behavioral changes.

Since our "internal" node crates are so inter-woven they're constantly making semver breaking changes.